src/sql/workbench/services/objectExplorer/browser/objectExplorerViewTreeShim.ts

```diff
@@ -111,42 +110,28 @@ export class OEShimService extends Disposable implements IOEShimService {
 	}
 
 	public async disconnectNode(viewId: string, node: ITreeItem): Promise<boolean> {
-		// we assume only nodes with payloads can be connected
-		// check to make sure we have an existing connection
-		let key = generateSessionMapKey(viewId, node);
-		let session = this.sessionMap.get(key);
-		if (session) {
-			let closed = (await this.oe.closeSession(node.childProvider!, this.oe.getSession(session)!))!.success;
-			if (closed) {
-				this.sessionMap.delete(key);
-			}
+		if (node.sessionId !== undefined) {
+			const closed = (await this.oe.closeSession(node.childProvider!, this.oe.getSession(node.sessionId!)!))!.success;
```

Comment:
`node.sessionId!` is unnecessary - you already know it's not undefined because of the if check